crates/sui-node/src/main.rs

```diff
@@ -82,6 +82,12 @@ fn main() {
         config.metrics_address
     );
 
+    #[cfg(feature = "gas-profiler")]
+    warn!(
```

Comment:
I think we should consider using [`std::compile_error`](https://doc.rust-lang.org/std/macro.compile_error.html#:~:text=Macro%20std%3A%3Acompile_error&text=Causes%20compilation%20to%20fail%20with,error%20messages%20for%20erroneous%20conditions.) for this.

Also note that it is not sufficient to perform this test in the `sui-node` crate, because what this is testing is whether you somehow explicitly enabled `gas-profiler` for `sui-node` which should never be true (`sui-node` shouldn't even have such a feature).

Instead, you will need to perform this test in the module containing the gas profiler code.  And you'll have to structure it slightly differently.  Namely, the validator and fullnode codebases would need to introduce an `authority` feature (or similar, maybe you want to call it `trusted` or `trusted-compute-base` which is more generic) that gets threaded down into the VM, and the gas profiler module should complain if it is enabled when the `authority`/`trusted`/`trusted-compute-base` feature is also enabled.

Reply:
In order to thread a feature from the authority to the vm, we would need an import chain in that direction, but it would most likely not be possible due to at some point causing circular imports. Nonetheless, I will try to see if I can find an import path that would work - if not, I am also attempting not having the feature as default even for the sui binary, and then adding the feature to the install command in the docs to enable it for only the CLI

Reply:
I solved this with a slightly different approach, we now will throw a compile error in the sui-node binary if the gas-profiler is enabled by checking a pair of macros defined in the gas profiler crate that will execute the code in the brackets if the feature is enabled, and do nothing if the feature is disabled. This is the approach the serde library uses to check if a feature is enabled in a different crate https://docs.rs/serde/1.0.186/serde/macro.serde_if_integer128.html

Reply:
edit: I had to remove the compile error and replace with a regular error, this is because [For the purpose of generating Cargo.lock, the resolver builds the dependency graph as-if all features of all workspace members are enabled.](https://doc.rust-lang.org/cargo/reference/resolver.html) which will always cause the compile error to be thrown, as evidenced by all my force push attempts to build the commit 09edc811d90505f58089f504554805c21b35ea83

Reply:
This is so unfortunate 😢 (but actually I kinda get why they did that, but it still doesn't stop me from being very sad about it).